src/Network/Network/VirtualNetwork/Subnet/ResetAzureVirtualNetworkSubnetPolicyCommand.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Reset, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualNetworkSubnetPolicy", SupportsShouldProcess = true, DefaultParameterSetName = "SetByResourceId"), OutputType(typeof(PSSubnet))]
+    public class ResetAzureVirtualNetworkSubnetPolicyCommand : VirtualNetworkBaseCmdlet
+    {
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "DefaultParameterSet",
+            HelpMessage = "The name of the subnet")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "SetByResourceId",
+            HelpMessage = "The name of the subnet")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "DefaultParameterSet",
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "SetByResourceId",
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Alias("InputObject")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = "DefaultParameterSet",
+            ValueFromPipeline = true,
+            HelpMessage = "The virtualNetwork")]
+        public PSVirtualNetwork VirtualNetwork { get; set; }
+
+        [Parameter(
+            ParameterSetName = "SetByResourceId",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id of the virtual network resource")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.ResourceId) || !string.IsNullOrWhiteSpace(this.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId.Trim('"'));
+                var resourceGroupName = resourceIdentifier.ResourceGroupName;
+                var virtualNetworkName = resourceIdentifier.ResourceName;
+                this.VirtualNetwork = GetVirtualNetwork(resourceGroupName, virtualNetworkName);
+            }
+
+            // Verify if the subnet exists in the VirtualNetwork
+            var subnet = this.VirtualNetwork.Subnets.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Is that possible VirtualNetwork or VirtualNetwork.Subnets be null?

Reply:
Added [ValidateNotNullOrEmpty]